dali/operators/input/input_operator_test.cu

```diff
@@ -0,0 +1,180 @@
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <thrust/host_vector.h>
+#include <thrust/device_vector.h>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+#include "dali/c_api.h"
+#include "dali/pipeline/pipeline.h"
+#include "dali/test/dali_test_config.h"
+
+namespace dali::test {
+
+
+struct InputOperatorMixedTestParam {
+  int cpu_queue_depth, gpu_queue_depth;
+  bool exec_pipelined, exec_async;
+  bool cpu_input;
+};
+
+
+namespace {
+
+InputOperatorMixedTestParam input_operator_test_params_simple_executor[] = {
+        {1, 1, false, false, true},
+        {2, 2, false, false, false},
+};
+
+
+InputOperatorMixedTestParam input_operator_test_params_pipelined_executor_uniform_queue[] = {
+        {2, 2, true, false, true},
+        {3, 3, true, false, true},
+        {2, 2, true, true,  false},
+        {3, 3, true, true,  false},
+};
+
+InputOperatorMixedTestParam input_operator_test_params_pipelined_executor_separate_queue[] = {
+        {2, 3, true, false, true},
+        {3, 2, true, false, true},
+        {2, 3, true, true,  false},
+        {3, 2, true, true,  false},
+};
+
+
+template<typename T>
+thrust::host_vector<T> random_vector_cpu(std::mt19937 &mt, size_t size) {
+  thrust::host_vector<T> cpu(size);
+  std::uniform_int_distribution<T> dist{0, 255};
+  auto gen = [&]() { return dist(mt); };
+  thrust::generate(cpu.begin(), cpu.end(), gen);
+  return cpu;
+}
+
+}  // namespace
+
+
+
+class InputOperatorMixedTest : public ::testing::TestWithParam<InputOperatorMixedTestParam> {
+ protected:
+  void SetUp() final {
+    auto parameters = GetParam();
+    cpu_queue_depth_ = parameters.cpu_queue_depth;
+    gpu_queue_depth_ = parameters.gpu_queue_depth;
+    exec_async_ = parameters.exec_async;
+    exec_pipelined_ = parameters.exec_pipelined;
+    exec_separated_ = cpu_queue_depth_ != gpu_queue_depth_;
+    cpu_input_ = parameters.cpu_input;
+    cout << "TEST: cpu_queue_depth=" << cpu_queue_depth_ << ", gpu_queue_depth=" << gpu_queue_depth_
+         << ", exec_async=" << std::boolalpha << exec_async_ << ", exec_pipelined="
+         << exec_pipelined_ << endl;
+
+    pipeline_ = std::make_unique<Pipeline>(batch_size_, num_threads_, device_id_, -1,
+                                           exec_pipelined_, cpu_queue_depth_, exec_async_);
+    pipeline_->SetExecutionTypes(exec_pipelined_, exec_separated_, exec_async_);
+
+    PutTogetherDaliGraph();
+
+    serialized_pipeline_ = pipeline_->SerializeToProtobuf();
+  }
+
+
+  int batch_size_ = 3;
+  int num_threads_ = 2;
+  int device_id_ = 0;
+  int n_iterations_ = 50;
+  std::unique_ptr<Pipeline> pipeline_;
+  std::string operator_name_ = "PTI";
+
+  bool exec_pipelined_, exec_async_, exec_separated_;
+  int cpu_queue_depth_, gpu_queue_depth_;
+  bool cpu_input_;
+  std::string serialized_pipeline_;
+
+ private:
+  void PutTogetherDaliGraph() {
+    pipeline_->AddOperator(OpSpec("PassthroughInput")
```

Comment:
Ideally this PassthroughInput would be defined in a _test.* file so that it doesn't belong to DALI library